midterm-2023-11-07/code/5311-kristijan-lazarov/script.ts

```diff
@@ -0,0 +1,322 @@
+    for (const movie of moives) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+
+}
+
+const generateMovieRow = (movie: Movie) => {
+    const row = document.createElement("div");
+    row.classList.add("movie-row");
+
+    //id
+    const idCell = document.createElement("div");
+    idCell.classList.add("movie-data", "movie-id");
+    idCell.innerHTML = movie.id.toString();
+    row.appendChild(idCell);
+
+
+    //name
+    const nameCell = document.createElement("div");
+    nameCell.classList.add("movie-data", "movie-name");
+    nameCell.innerHTML = movie.title;
+    row.appendChild(nameCell);
+
+
+    //director
+    const directorCell = document.createElement("div");
+    directorCell.classList.add("movie-data", "movie-director");
+    directorCell.innerHTML = movie.director;
+    row.appendChild(directorCell);
+
+
+    //year
+    const relaseCell = document.createElement("div");
+    relaseCell.classList.add("movie-data", "movie-relese");
+    relaseCell.innerHTML = movie.year.toString();
+    row.appendChild(relaseCell);
+
+
+    //genre
+    const genreCell = document.createElement("div");
+    genreCell.classList.add("movie-data", "movie-genre");
+    genreCell.innerHTML = movie.genre.join(" / ");
+    row.appendChild(genreCell);
+
+
+    //plot
+    const plotCell = document.createElement("div");
+    plotCell.classList.add("movie-data", "movie-plot");
+    const maxCharacters = 50;
+    let content = movie.plot;
+
+    if (content.length > maxCharacters) {
+        const lastword = content.lastIndexOf(" ", maxCharacters);
+        content = content.substring(0, lastword) + "...";
+    }
+    plotCell.innerHTML = content;
+
+    plotCell.addEventListener("click", () => {
+        const modal = document.getElementById("plot-details")!;
+        modal.classList.remove("hidden");
+
+        const plotParagraph = document.getElementById("movie-details-plot")!;
+        plotParagraph.innerText = movie.plot;
+    });
+
+    row.appendChild(plotCell);
+
+    //actors
+    const actorsCell = document.createElement("div");
+    actorsCell.classList.add("movie-data", "movie-actors");
+    const actorList = document.createElement("ul");
+    actorList.id = "movie-details-actor-list"
+    for (const actor of movie.cast) {
+        const actorItem = document.createElement("li");
+        actorItem.innerText = `${actor.actor} (${actor.character})`;
+        actorList.appendChild(actorItem);
+    }
+    actorsCell.appendChild(actorList);
+    row.appendChild(actorsCell);
+
+
+    //oscars
+    const oscarsCell = document.createElement("div");
+    oscarsCell.classList.add("movie-data", "movie-oscars");
+
+    if (movie.oscars) {
+        const numOscars = Object.values(movie.oscars).length;
+        oscarsCell.innerHTML = numOscars.toString();
+    } else {
+        oscarsCell.innerHTML = "0";
+    }
+
+    row.appendChild(oscarsCell);
+    return row;
+}
+
+if (document) {
```

Comment:
This can also be done in the `DOMContentLoaded` handler - this way is functional, but a bit outdated.